Processing recipe rework

fscan opened the pull request:
Work in progress!

Processing recipes based on json recipe parsing.

*) Removed recipes for external mods for now. If we really need it we can add it back easy enough.
*) Removed WebSerializer. This probably has to be reimplemented.
*) The old recipe parser still has some stuff in the API

orod-org commented on src/main/java/appeng/recipes/AERecipeLoader.java:
```diff
@@ -0,0 +1,121 @@
+	private boolean process( final Path root, final Path file )
+	{
+		String relative = root.relativize( file ).toString();
+		if( !"json".equals( FilenameUtils.getExtension( file.toString() ) ) || relative.startsWith( "_" ) )
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00121)

orod-org commented on src/main/java/appeng/recipes/AERecipeLoader.java:
```diff
@@ -0,0 +1,121 @@
+
+	private void register( JsonObject json )
+	{
+		if( json == null || json.isJsonNull() )
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00121)

orod-org commented on src/main/java/appeng/recipes/AERecipeLoader.java:
```diff
@@ -0,0 +1,121 @@
+public class AERecipeLoader
+{
+	private static final String AERECIPE_BASE = "/aerecipes";
+	private static Gson GSON = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this field "GSON" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3008)

orod-org commented on src/main/java/appeng/recipes/AERecipeLoader.java:
```diff
@@ -0,0 +1,121 @@
+		return CraftingHelper.findFiles( this.mod, "assets/" + AppEng.MOD_ID + AERECIPE_BASE, this::preprocess, this::process, true, true );
+	}
+
+	private boolean preprocess( final Path root )
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "root". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1172)

orod-org commented on src/main/java/appeng/recipes/AERecipeLoader.java:
```diff
@@ -0,0 +1,121 @@
+			throw new JsonSyntaxException( "Json cannot be null" );
+
+		String type = this.ctx.appendModId( JsonUtils.getString( json, "type" ) );
+		if( type.isEmpty() )
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00121)

orod-org commented on src/main/java/appeng/recipes/AERecipeLoader.java:
```diff
@@ -0,0 +1,121 @@
+		{
+			reader = Files.newBufferedReader( file );
+			JsonObject json = JsonUtils.fromJson( GSON, reader, JsonObject.class );
+			if( json.has( "conditions" ) && !CraftingHelper.processConditions( JsonUtils.getJsonArray( json, "conditions" ), this.ctx ) )
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00121)

orod-org commented on src/main/java/appeng/recipes/AERecipeLoader.java:
```diff
@@ -0,0 +1,121 @@
+			throw new JsonSyntaxException( "Recipe type can not be an empty string" );
+
+		IAERecipeFactory factory = factories.get( new ResourceLocation( type ) );
+		if( factory == null )
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00121)

orod-org commented on src/main/java/appeng/core/Registration.java:
```diff
@@ -225,26 +201,13 @@ private void registerSpatialDimension()
-		Preconditions.checkNotNull( customRecipeConfig );
 
 		final Api api = Api.INSTANCE;
 		final IPartHelper partHelper = api.partHelper();
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "partHelper". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "partHelper" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1481)

orod-org commented on src/main/java/appeng/recipes/handlers/GrinderHandler.java:
```diff
@@ -0,0 +1,49 @@
+	@Override
+	public void register( JsonObject json, JsonContext ctx )
+	{
+		// TODO only primary for now
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1135)

orod-org commented on src/main/java/appeng/recipes/IAERecipeFactory.java:
```diff
@@ -0,0 +1,13 @@
+import net.minecraftforge.common.crafting.JsonContext;
+
+
+public interface IAERecipeFactory
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Annotate the "IAERecipeFactory" interface with the @FunctionalInterface annotation [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1609)

orod-org commented:
SonarQube analysis reported 70 issues
* ![BLOCKER][BLOCKER] 5 blocker
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 50 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [AERecipeLoader.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/29a3d85e427879ddba61d7719d8cc519455d3ad2/src/main/java/appeng/recipes/AERecipeLoader.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [IAERecipeFactory.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/29a3d85e427879ddba61d7719d8cc519455d3ad2/src/main/java/appeng/recipes/IAERecipeFactory.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [GrinderHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/29a3d85e427879ddba61d7719d8cc519455d3ad2/src/main/java/appeng/recipes/handlers/GrinderHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [InscriberHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/29a3d85e427879ddba61d7719d8cc519455d3ad2/src/main/java/appeng/recipes/handlers/InscriberHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [SmeltingHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/29a3d85e427879ddba61d7719d8cc519455d3ad2/src/main/java/appeng/recipes/handlers/SmeltingHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L289](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/29a3d85e427879ddba61d7719d8cc519455d3ad2/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L289): Define a constant instead of duplicating this literal "sky_stone_chest" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L348](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/29a3d85e427879ddba61d7719d8cc519455d3ad2/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L348): Define a constant instead of duplicating this literal "quantum_ring" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L432](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/29a3d85e427879ddba61d7719d8cc519455d3ad2/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L432): Define a constant instead of duplicating this literal "crafting_unit" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L444](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/29a3d85e427879ddba61d7719d8cc519455d3ad2/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L444): Define a constant instead of duplicating this literal "crafting_storage" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [PartType.java#L229](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/29a3d85e427879ddba61d7719d8cc519455d3ad2/src/main/java/appeng/items/parts/PartType.java#L229): Define a constant instead of duplicating this literal "p2p_tunnel" 7 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

orod-org commented on src/api/java/appeng/api/recipes/IIngredient.java:
```diff
@@ -29,7 +29,7 @@
 
-
+@Deprecated
 public interface IIngredient
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

orod-org commented on src/main/java/appeng/core/Registration.java:
```diff
@@ -225,26 +201,13 @@ private void registerSpatialDimension()
 	private void registerCraftHandlers( final IRecipeHandlerRegistry registry )
 	{
 		registry.addNewSubItemResolver( new AEItemResolver() );
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "addNewSubItemResolver"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

orod-org commented on src/api/java/appeng/api/recipes/ICraftHandler.java:
```diff
@@ -30,7 +30,7 @@
 
-
+@Deprecated
 public interface ICraftHandler
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

orod-org commented on src/api/java/appeng/api/features/IRecipeHandlerRegistry.java:
```diff
@@ -48,6 +48,7 @@
 	 * @param handler class of crafthandler
 	 */
+	@Deprecated
 	void addNewCraftHandler( String name, Class<? extends ICraftHandler> handler );
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

orod-org commented on src/api/java/appeng/api/features/IRecipeHandlerRegistry.java:
```diff
@@ -65,11 +67,13 @@
 	 */
 	@Nullable
+	@Deprecated
 	ICraftHandler getCraftHandlerFor( String name );
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

orod-org commented on src/api/java/appeng/api/features/IRecipeHandlerRegistry.java:
```diff
@@ -65,11 +67,13 @@
 	 * @return a new recipe handler, which can be used to parse, and read recipe files.
 	 */
+	@Deprecated
 	IRecipeHandler createNewRecipehandler();
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

orod-org commented on src/api/java/appeng/api/features/IRecipeHandlerRegistry.java:
```diff
@@ -57,6 +58,7 @@
 	 * @param sir sub item resolver
 	 */
+	@Deprecated
 	void addNewSubItemResolver( ISubItemResolver sir );
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

orod-org commented on src/api/java/appeng/api/recipes/IRecipeLoader.java:
```diff
@@ -28,7 +28,7 @@
 
-
+@Deprecated
 public interface IRecipeLoader
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Annotate the "IRecipeLoader" interface with the @FunctionalInterface annotation [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1609)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

orod-org commented on src/main/java/appeng/recipes/RecipeHandler.java:
```diff
@@ -74,696 +31,15 @@
  */
 public class RecipeHandler implements IRecipeHandler
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "IRecipeHandler"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

orod-org commented on src/main/java/appeng/recipes/RecipeHandler.java:
```diff
@@ -74,696 +31,15 @@
-	}
-
 	@Override
 	public void parseRecipes( final IRecipeLoader loader, final String path )
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "IRecipeLoader"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

orod-org commented on src/api/java/appeng/api/recipes/IRecipeHandler.java:
```diff
@@ -27,6 +27,7 @@
  * Represents the AE2 Recipe Loading/Reading Class
  */
+@Deprecated
 public interface IRecipeHandler
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

orod-org commented:
SonarQube analysis reported 95 issues
* ![BLOCKER][BLOCKER] 5 blocker
* ![CRITICAL][CRITICAL] 8 critical
* ![MAJOR][MAJOR] 19 major
* ![MINOR][MINOR] 53 minor
* ![INFO][INFO] 10 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [AERecipeLoader.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d706521940043b97c964f7c1dacb7551ac1b328b/src/main/java/appeng/recipes/AERecipeLoader.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [IAERecipeFactory.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d706521940043b97c964f7c1dacb7551ac1b328b/src/main/java/appeng/recipes/IAERecipeFactory.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [GrinderHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d706521940043b97c964f7c1dacb7551ac1b328b/src/main/java/appeng/recipes/handlers/GrinderHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [InscriberHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d706521940043b97c964f7c1dacb7551ac1b328b/src/main/java/appeng/recipes/handlers/InscriberHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [SmeltingHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d706521940043b97c964f7c1dacb7551ac1b328b/src/main/java/appeng/recipes/handlers/SmeltingHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L289](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d706521940043b97c964f7c1dacb7551ac1b328b/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L289): Define a constant instead of duplicating this literal "sky_stone_chest" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L348](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d706521940043b97c964f7c1dacb7551ac1b328b/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L348): Define a constant instead of duplicating this literal "quantum_ring" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L432](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d706521940043b97c964f7c1dacb7551ac1b328b/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L432): Define a constant instead of duplicating this literal "crafting_unit" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L444](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d706521940043b97c964f7c1dacb7551ac1b328b/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L444): Define a constant instead of duplicating this literal "crafting_storage" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [PartType.java#L229](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d706521940043b97c964f7c1dacb7551ac1b328b/src/main/java/appeng/items/parts/PartType.java#L229): Define a constant instead of duplicating this literal "p2p_tunnel" 7 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

orod-org commented:
SonarQube analysis reported 95 issues
* ![BLOCKER][BLOCKER] 5 blocker
* ![CRITICAL][CRITICAL] 8 critical
* ![MAJOR][MAJOR] 19 major
* ![MINOR][MINOR] 53 minor
* ![INFO][INFO] 10 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [AERecipeLoader.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c7a5b1e5d2303ae64e24ab19c3c7f2ac89e6a27d/src/main/java/appeng/recipes/AERecipeLoader.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [IAERecipeFactory.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c7a5b1e5d2303ae64e24ab19c3c7f2ac89e6a27d/src/main/java/appeng/recipes/IAERecipeFactory.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [GrinderHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c7a5b1e5d2303ae64e24ab19c3c7f2ac89e6a27d/src/main/java/appeng/recipes/handlers/GrinderHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [InscriberHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c7a5b1e5d2303ae64e24ab19c3c7f2ac89e6a27d/src/main/java/appeng/recipes/handlers/InscriberHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [SmeltingHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c7a5b1e5d2303ae64e24ab19c3c7f2ac89e6a27d/src/main/java/appeng/recipes/handlers/SmeltingHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L289](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c7a5b1e5d2303ae64e24ab19c3c7f2ac89e6a27d/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L289): Define a constant instead of duplicating this literal "sky_stone_chest" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L348](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c7a5b1e5d2303ae64e24ab19c3c7f2ac89e6a27d/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L348): Define a constant instead of duplicating this literal "quantum_ring" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L432](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c7a5b1e5d2303ae64e24ab19c3c7f2ac89e6a27d/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L432): Define a constant instead of duplicating this literal "crafting_unit" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L444](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c7a5b1e5d2303ae64e24ab19c3c7f2ac89e6a27d/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L444): Define a constant instead of duplicating this literal "crafting_storage" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [PartType.java#L229](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c7a5b1e5d2303ae64e24ab19c3c7f2ac89e6a27d/src/main/java/appeng/items/parts/PartType.java#L229): Define a constant instead of duplicating this literal "p2p_tunnel" 7 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

orod-org commented:
SonarQube analysis reported 95 issues
* ![BLOCKER][BLOCKER] 5 blocker
* ![CRITICAL][CRITICAL] 8 critical
* ![MAJOR][MAJOR] 19 major
* ![MINOR][MINOR] 53 minor
* ![INFO][INFO] 10 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [AERecipeLoader.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/26f94cf92df1c3f8de424fe7dad901f8ec89ab7f/src/main/java/appeng/recipes/AERecipeLoader.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [IAERecipeFactory.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/26f94cf92df1c3f8de424fe7dad901f8ec89ab7f/src/main/java/appeng/recipes/IAERecipeFactory.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [GrinderHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/26f94cf92df1c3f8de424fe7dad901f8ec89ab7f/src/main/java/appeng/recipes/handlers/GrinderHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [InscriberHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/26f94cf92df1c3f8de424fe7dad901f8ec89ab7f/src/main/java/appeng/recipes/handlers/InscriberHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [SmeltingHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/26f94cf92df1c3f8de424fe7dad901f8ec89ab7f/src/main/java/appeng/recipes/handlers/SmeltingHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L289](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/26f94cf92df1c3f8de424fe7dad901f8ec89ab7f/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L289): Define a constant instead of duplicating this literal "sky_stone_chest" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L348](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/26f94cf92df1c3f8de424fe7dad901f8ec89ab7f/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L348): Define a constant instead of duplicating this literal "quantum_ring" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L432](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/26f94cf92df1c3f8de424fe7dad901f8ec89ab7f/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L432): Define a constant instead of duplicating this literal "crafting_unit" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L444](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/26f94cf92df1c3f8de424fe7dad901f8ec89ab7f/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L444): Define a constant instead of duplicating this literal "crafting_storage" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [PartType.java#L229](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/26f94cf92df1c3f8de424fe7dad901f8ec89ab7f/src/main/java/appeng/items/parts/PartType.java#L229): Define a constant instead of duplicating this literal "p2p_tunnel" 7 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

orod-org commented:
SonarQube analysis reported 92 issues
* ![BLOCKER][BLOCKER] 5 blocker
* ![CRITICAL][CRITICAL] 8 critical
* ![MAJOR][MAJOR] 18 major
* ![MINOR][MINOR] 52 minor
* ![INFO][INFO] 9 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [AERecipeLoader.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/recipes/AERecipeLoader.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [IAERecipeFactory.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/recipes/IAERecipeFactory.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [GrinderHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/recipes/handlers/GrinderHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [InscriberHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/recipes/handlers/InscriberHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [SmeltingHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/recipes/handlers/SmeltingHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L289](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L289): Define a constant instead of duplicating this literal "sky_stone_chest" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L348](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L348): Define a constant instead of duplicating this literal "quantum_ring" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L432](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L432): Define a constant instead of duplicating this literal "crafting_unit" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L444](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L444): Define a constant instead of duplicating this literal "crafting_storage" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [PartType.java#L229](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/items/parts/PartType.java#L229): Define a constant instead of duplicating this literal "p2p_tunnel" 7 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

orod-org commented on src/main/java/appeng/items/parts/PartType.java:
```diff
@@ -183,11 +183,12 @@ public boolean isCable()
 	QUARTZ_FIBER( 140, "quartz_fiber", EnumSet.of( AEFeature.QUARTZ_FIBER ), EnumSet.noneOf( IntegrationType.class ), PartQuartzFiber.class ),
 
-	MONITOR( 160, "monitor", EnumSet.of( AEFeature.PANELS ), EnumSet.noneOf( IntegrationType.class ), PartPanel.class ),
+	MONITOR( 160, "monitor", EnumSet.of( AEFeature.PANELS ), EnumSet.noneOf( IntegrationType.class ), PartPanel.class, "itemIlluminatedPanel" ),
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "itemIlluminatedPanel" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)

orod-org commented:
SonarQube analysis reported 99 issues
* ![BLOCKER][BLOCKER] 5 blocker
* ![CRITICAL][CRITICAL] 12 critical
* ![MAJOR][MAJOR] 21 major
* ![MINOR][MINOR] 52 minor
* ![INFO][INFO] 9 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [AERecipeLoader.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/recipes/AERecipeLoader.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [IAERecipeFactory.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/recipes/IAERecipeFactory.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [GrinderHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/recipes/handlers/GrinderHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [InscriberHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/recipes/handlers/InscriberHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [SmeltingHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/recipes/handlers/SmeltingHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L289](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L289): Define a constant instead of duplicating this literal "sky_stone_chest" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L348](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L348): Define a constant instead of duplicating this literal "quantum_ring" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L432](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L432): Define a constant instead of duplicating this literal "crafting_unit" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L444](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L444): Define a constant instead of duplicating this literal "crafting_storage" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ItemPart.java#L247](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/0ce4e38e38f2609842cdf5476522b7cd7cc8d1f0/src/main/java/appeng/items/parts/ItemPart.java#L247): Define a constant instead of duplicating this literal "Unable to construct IBusPart from IBusItem : " 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

fscan commented:
Rebased (and squashed to make it easier).
We still need to take a closer look at the exposed API. As this is now for rv6 only, changes are possible.

orod-org commented:
SonarQube analysis reported 79 issues
* ![BLOCKER][BLOCKER] 5 blocker
* ![CRITICAL][CRITICAL] 11 critical
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 52 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [AERecipeLoader.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8360ec978259c34ab1ecb16a989345c0f53a7d5e/src/main/java/appeng/recipes/AERecipeLoader.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [IAERecipeFactory.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8360ec978259c34ab1ecb16a989345c0f53a7d5e/src/main/java/appeng/recipes/IAERecipeFactory.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [GrinderHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8360ec978259c34ab1ecb16a989345c0f53a7d5e/src/main/java/appeng/recipes/handlers/GrinderHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [InscriberHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8360ec978259c34ab1ecb16a989345c0f53a7d5e/src/main/java/appeng/recipes/handlers/InscriberHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![BLOCKER][BLOCKER] [SmeltingHandler.java#](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8360ec978259c34ab1ecb16a989345c0f53a7d5e/src/main/java/appeng/recipes/handlers/SmeltingHandler.java#): Add or update the header of this file. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1451)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L292](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8360ec978259c34ab1ecb16a989345c0f53a7d5e/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L292): Define a constant instead of duplicating this literal "sky_stone_chest" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L351](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8360ec978259c34ab1ecb16a989345c0f53a7d5e/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L351): Define a constant instead of duplicating this literal "quantum_ring" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L435](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8360ec978259c34ab1ecb16a989345c0f53a7d5e/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L435): Define a constant instead of duplicating this literal "crafting_unit" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ApiBlocks.java#L447](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8360ec978259c34ab1ecb16a989345c0f53a7d5e/src/main/java/appeng/core/api/definitions/ApiBlocks.java#L447): Define a constant instead of duplicating this literal "crafting_storage" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ItemPart.java#L247](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8360ec978259c34ab1ecb16a989345c0f53a7d5e/src/main/java/appeng/items/parts/ItemPart.java#L247): Define a constant instead of duplicating this literal "Unable to construct IBusPart from IBusItem : " 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
